lib/ansible/plugins/callback/__init__.py

```diff
@@ -299,12 +299,13 @@ def _handle_exception(self, result, use_stderr=False):
 
         if 'exception' in result:
             msg = "An exception occurred during task execution. "
+            exception_str = to_text(result['exception'])
```

Comment:
this conversion should not be done here, but when the result is first processed ... also I would consider it a bug of the module emitting such a result

Reply:
Since this is a function that's directly called by other callback plugins, I don't see how this can be achieved except by convention. And preventing crashes by hoping that all parts (including a lot of code not part of this repository) sticks to convention is a bit extreme IMO.